src/sentry/static/sentry/app/components/events/interfaces/debugMeta/debugFileDetails/uploadedDebugFilesTable.tsx

```diff
@@ -0,0 +1,155 @@
+      }
+    );
+  };
+
+  getCandidates() {
+    const {debugFiles} = this.state;
+    const {candidates} = this.props;
+
+    return candidates.map(candidate => {
+      if (!(debugFiles ?? []).find(debugFile => debugFile.id === candidate.location)) {
+        return {
+          ...candidate,
+          download: {
+            status: CandidateDownloadStatus.DELETED,
+            details: t('This file was deleted'),
+          },
+        };
+      }
+      return candidate;
+    }) as Candidates;
+  }
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderBody() {
+    const {organization, projectId} = this.props;
+    const {loading} = this.state;
+
+    const candidates = this.getCandidates();
+
+    return (
+      <Table
+        title={t('Uploaded Debug Files')}
+        description="this is a description"
+        candidates={candidates}
+        emptyMessage={t("You haven't uploaded any debug file")}
+        isLoading={loading}
+        actions={({debugId, deleted}) => {
+          const actions = (
+            <ButtonBar gap={0.5}>
+              <Role role={organization.debugFilesRole} organization={organization}>
+                {({hasRole}) => (
+                  <Tooltip
+                    disabled={hasRole}
+                    title={t('You do not have permission to download debug files.')}
+                  >
+                    <Button
+                      size="xsmall"
+                      icon={<IconDownload size="xs" />}
+                      href={`${this.api.baseUrl}/projects/${organization.slug}/${projectId}/files/dsyms/?id=${debugId}`}
+                      disabled={!hasRole || deleted}
+                    >
+                      {t('Download')}
+                    </Button>
+                  </Tooltip>
+                )}
+              </Role>
+              <Access access={['project:write']} organization={organization}>
+                {({hasAccess}) => (
+                  <Tooltip
+                    disabled={hasAccess}
+                    title={t('You do not have permission to delete debug files.')}
+                  >
+                    <Confirm
+                      confirmText={t('Delete')}
+                      message={t('Are you sure you wish to delete this file?')}
+                      onConfirm={() => this.handleDelete(debugId)}
+                      disabled={!hasAccess || deleted}
+                    >
+                      <Button
+                        priority="danger"
+                        icon={<IconDelete size="xs" />}
+                        size="xsmall"
+                        disabled={!hasAccess || deleted}
+                      />
+                    </Confirm>
+                  </Tooltip>
+                )}
+              </Access>
+            </ButtonBar>
+          );
+
+          if (!deleted) {
+            return actions;
+          }
+
+          return <Tooltip title={t('Actions not available.')}>{actions}</Tooltip>;
```

Comment:
fyi you can use the `disabled` prop on `Tooltip`

```suggestion
          return <Tooltip disabled={!deleted} title={t('Actions not available.')}>{actions}</Tooltip>;
```

Reply:
I did not want to render the tooltip component if not necessary